### ISSUE TITLE
Graphql endpoint output and API KEY on lambda function not being passed properly

### ISSUE THREAD

**Issue by aprilmintacpineda** [aws-amplify/amplify-cli#5143]
>**Note**: If your issue/bug is regarding the AWS Amplify Console service, please log it in the 
[Amplify Console GitHub Issue Tracker](https://github.com/aws-amplify/amplify-console/issues)

**Describe the bug**

I created a function and gave it permission to the API but when I do `console.log(process.env)` I'm getting these values which weren't what I was expecting, I was expecting them to hold the actual values such as HTTPS link to graphql endpoint and the actual API Key as indicated here on this tutorial https://docs.amplify.aws/lib/graphqlapi/graphql-from-nodejs/q/platform/js#mutation

![image](https://user-images.githubusercontent.com/21032419/90864665-ac799b00-e3c3-11ea-9ed0-403469d5bb70.png)

I created my function like `amplify add function` and then said `y` when asked `do you want to access other resources in this function` and then I put in API and selected the table/model I want to update in this function (it only needs one table).

I have the following code in a lambda layer being used by that function:

```js
// libs/graphqlClient.js

const { GraphQLClient, gql } = require('graphql-request');

const graphqlClient = new GraphQLClient(
  process.env.API_LEADSMANAGEMENTSYS_GRAPHQLAPIENDPOINTOUTPUT,
  {
    headers: {
      'x-api-key': process.env.API_LEADSMANAGEMENTSYS_GRAPHQLAPIKEYOUTPUT
    }
  }
);

module.exports.graphqlClient = graphqlClient;
module.exports.gql = gql;
```

but when I try to do

```js
const { graphqlClient, gql } = require('/opt/libs/graphqlClient');

async function handler (event) {
  console.log(JSON.stringify(event, null, 2));

  const query = gql`
    {
      listUsers {
        items {
          id
          firstName
          middleName
          lastName
        }
      }
    }
  `;
}
```

It will throw an error complaining about the URL because the `process.env.API_LEADSMANAGEMENTSYS_GRAPHQLAPIKEYOUTPUT` is not a valid URL.

**Amplify CLI Version**

always "latest" as specified in the CI/CD, my local amplify is 4.27.2

**To Reproduce**

Just follow what was said above

**Expected behavior**

`process.env.API_LEADSMANAGEMENTSYS_GRAPHQLAPIKEYOUTPUT` should contain the actual URL of the graphql endpoint as indicated on the example on the docs.

**Screenshots**

**Desktop (please complete the following information):**
 - OS: N/A
 - Node Version: N/A

**Additional context**

N/A

**Comment by aprilmintacpineda** [comment]
@yuth Why is this pending response and not-reproducible?

**Comment by yuth** [comment]
I could not reproduce this. I was wondering if you could share the Cloudformation snippet of the lambda function

**Comment by aprilmintacpineda** [comment]
I actually encountered this while doing my online course with aws-amplify and react-js
https://www.udemy.com/course/build-it-with-me-lead-management-system-web-app

**Comment by am29d** [comment]
Can confirm, created a function via CLI, when asked if I want to have access to API, I selected yes. The `dependsOn` block is an empty array. I had created a function few month ago in similar project and it worked The dependsOn block looked like this:
```json
"dependsOn": [
        {
          "category": "api",
          "resourceName": "xxxxxxGraphQlAPI",
          "attributes": [
            "GraphQLAPIIdOutput",
            "GraphQLAPIEndpointOutput"
          ]
        }
      ]
```

Amplify version is `4.29.2`.

**Comment by MagneH** [comment]
I experienced the same. I had to change the default auth on my api to api key. You can still choose to configure Cognito User pools as a second method. Doing this and recreating the function made amplify pass the API key to the process.env object.

**Comment by aprilmintacpineda** [comment]
On previous versions of amplify, the default doesn't need to be the API KEY though, and I think API KEY should really be secondary method...

**Comment by UnleashedMind** [comment]
I just tried and I can not reproduce the issue with the latest CLI version. 
In the `amplify function add` workflow, I selected to access the api, and all the env variables 
are properly set in the Lambda function. 
API_<resource-name>_GRAPHQLAPIENDPOINTOUTPUT
API_<resource-name>_GRAPHQLAPIIDOUTPUT
API_<resource-name>_GRAPHQLAPIKEYOUTPUT

By the way, for the Lambda Layer, `opt` folder is meant to contain static contents, https://docs.amplify.aws/cli/function/layers#add-shared-code--assets

**Comment by cybercussion** [comment]
I just ran into this.

Repro - I created a Storage with Lambda Trigger.
I decided I'd add DynamoDB Table.  I ran into similar policy names created.  I manually prefixed them for S3 and DynamoDB so both policies could exist.
I found out editing dynamo doesn't event stream back to the UI, so I pivoted to GraphQL.
I got thru a painful 3 hour tour of getting axios, graphQL querying setup and finally not crashing the Lambda.
And the process.env.API_ASSETAPI_GRAPHQLAPIENDPOINTOUTPUT is defaulted to apiassetAPIGraphQLAPIEndpointOutput.

I don't know where the argument comes from but I know its in the aws-exports.js ok.
